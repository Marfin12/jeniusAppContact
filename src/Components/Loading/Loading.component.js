import * as React from 'react';
import { Modal, View, ActivityIndicator } from 'react-native';
import { useSelector } from 'react-redux';

import styles from './Loading.styles';
import config from './Loading.config';

const Loading = (props) => (
  <Modal transparent={true} animationType={'none'} visible={props.isVisible}>
    <View style={styles.container}>
      <ActivityIndicator size="large" color="#000000"/>
    </View>
  </Modal>
);

Loading.displayName = config.displayName;
Loading.defaultProps = config.defaultProps;

export default Loading;
