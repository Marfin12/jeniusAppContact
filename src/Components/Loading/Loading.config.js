const defaultProps = {
  isVisible: false
};

const displayName = 'Loading';

export default {
  defaultProps,
  displayName,
};
