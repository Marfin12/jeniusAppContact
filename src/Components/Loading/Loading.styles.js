import colors from '../../Assets/Colors';

const styles = {
  container: {
    flex: 1,
    justifyContent: "center",
    backgroundColor:'rgba(4,0,0,0.8)',
    zIndex: 1000
  }
};

export default styles;
