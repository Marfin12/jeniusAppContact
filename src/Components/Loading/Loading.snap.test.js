import React from 'react';
import renderer from 'react-test-renderer';

import Loading from './Loading.component';
import config from './Loading.config';

jest.mock('react-redux', () => {
  const ActualReactRedux = jest.requireActual('react-redux');
  const lightMode = {
    theme: false,
  };
  const darkMode = {
    theme: true,
  };
  return {
    ...ActualReactRedux,
    useSelector: jest
      .fn()
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => lightMode)
      .mockImplementationOnce(() => darkMode)
      .mockImplementationOnce(() => darkMode),
  };
});

describe('Light Mode Text Field Component', () => {
  test('renders text field component with default props', () => {
    const ComponentWrapper = renderer
      .create(<Loading {...config.defaultProps} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
});

describe('Dark Mode Text Field Component', () => {
  test('renders text field component with default props', () => {
    const ComponentWrapper = renderer
      .create(<Loading {...config.defaultProps} />)
      .toJSON();
    expect(ComponentWrapper).toMatchSnapshot();
  });
});
