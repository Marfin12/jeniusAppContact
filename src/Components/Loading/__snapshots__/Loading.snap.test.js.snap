// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dark Mode Text Field Component renders text field component with default props 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(4,0,0,0.8)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 1000,
      }
    }
  >
    <ActivityIndicator
      color="#000000"
      size="large"
    />
  </View>
</Modal>
`;

exports[`Light Mode Text Field Component renders text field component with default props 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  transparent={true}
  visible={false}
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(4,0,0,0.8)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 1000,
      }
    }
  >
    <ActivityIndicator
      color="#000000"
      size="large"
    />
  </View>
</Modal>
`;
